cnf-certification-test/certification/certtool/certtool_test.go

```diff
@@ -15,3 +15,170 @@
 // 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 package certtool
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/test-network-function/cnf-certification-test/internal/api"
+	"github.com/test-network-function/cnf-certification-test/pkg/configuration"
+)
+
+func TestGetContainerCertificationRequestFunction(t *testing.T) {
+	// Mock out the CertAPIClient
+	defer func() {
+		CertAPIClient = api.CertAPIClient{}
+	}()
+	CertAPIClient = &api.CertAPIClientFuncsMock{
+		GetContainerCatalogEntryFunc: func(id configuration.ContainerImageIdentifier) (*api.ContainerCatalogEntry, error) {
+			return &api.ContainerCatalogEntry{
+				ID: id.Name,
+			}, nil
+		},
+	}
+
+	testCases := []struct {
+		imageID configuration.ContainerImageIdentifier
+	}{
+		{
+			imageID: configuration.ContainerImageIdentifier{
+				Name:       "testID",
+				Repository: "testRepo",
+				Tag:        "testTag",
+				Digest:     "testDigest",
+			},
+		},
+	}
+
+	// Run the test
+	for _, tc := range testCases {
+		result := GetContainerCertificationRequestFunction(tc.imageID)
+		id, err := result()
+		assert.Nil(t, err)
+		myResult := id.(*api.ContainerCatalogEntry)
+		myResult.ID = tc.imageID.Name
```

Comment:
Shouldn't this be an assert?

Reply:
Yes, correct.  Not sure what I was thinking but it should have been an assert.  😅  Fixed.